src/com/cloudogu/ces/cesbuildlib/Trivy.groovy

```diff
@@ -0,0 +1,133 @@
+     * @return Returns true if the scan was ok (no vulnerability found); returns false if any vulnerability was found
+     */
+    boolean scanImage(
+        String imageName,
+        String severityLevel = TrivySeverityLevel.CRITICAL,
+        String strategy = TrivyScanStrategy.UNSTABLE,
+        // Avoid rate limits of default Trivy database source
+        String additionalFlags = "--db-repository public.ecr.aws/aquasecurity/trivy-db --java-db-repository public.ecr.aws/aquasecurity/trivy-java-db",
+        String trivyReportFile = "trivy/trivyReport.json"
+    ) {
+        Integer exitCode = docker.image("aquasec/trivy:${trivyVersion}")
+            .mountJenkinsUser()
+            .mountDockerSocket()
+            .inside("-v ${script.env.WORKSPACE}/.trivy/.cache:/root/.cache/") {
+                // Write result to $trivyReportFile in json format (--format json), which can be converted in the saveFormattedTrivyReport function
+                // Exit with exit code 10 if vulnerabilities are found or os is so old that trivy has no records for it anymore
+                script.sh("mkdir -p " + trivyDirectory)
+                script.sh(script: "trivy image --exit-code 10 --exit-on-eol 10 --format ${TrivyScanFormat.JSON} -o ${trivyReportFile} --severity ${severityLevel} ${additionalFlags} ${imageName}", returnStatus: true)
+            }
+        switch (exitCode) {
+            case 0:
+                // Everything all right, no vulnerabilities
+                return true
+            case 10:
+                // Found vulnerabilities
+                // Set build status according to strategy
+                switch (strategy) {
+                    case TrivyScanStrategy.IGNORE:
+                        break
+                    case TrivyScanStrategy.UNSTABLE:
+                        script.archiveArtifacts artifacts: "${trivyReportFile}", allowEmptyArchive: true
+                        script.unstable("Trivy has found vulnerabilities in image " + imageName + ". See " + trivyReportFile)
+                        break
+                    case TrivyScanStrategy.FAIL:
+                        script.archiveArtifacts artifacts: "${trivyReportFile}", allowEmptyArchive: true
+                        script.error("Trivy has found vulnerabilities in image " + imageName + ". See " + trivyReportFile)
+                        break
+                }
+                return false
+            default:
+                script.error("Error during trivy scan; exit code: " + exitCode)
+        }
+    }
+
+    /**
+     * Scans a dogu image for vulnerabilities.
+     * Notes:
+     * - Use a .trivyignore file for allowed CVEs
+     * - This function will generate a JSON formatted report file which can be converted to other formats via saveFormattedTrivyReport()
+     * - Evaluate via exit codes: 0 = no vulnerability; 1 = vulnerabilities found; other = function call failed
```

Comment:
see below. Should be exit code 10 instead of 1, shouldn't it?
```suggestion
     * - Evaluate via exit codes: 0 = no vulnerability; 10 = vulnerabilities found; other = function call failed
```

Reply:
This seems to be an old comment which doesn't apply any more as the function now has a boolean return value

Reply:
Done, see https://github.com/cloudogu/ces-build-lib/commit/cd604e834ad121321c5eda5362a42d62bccdcb15